src/Bicep.Cli/CommandLine/ArgumentParser.cs

```diff
@@ -2,30 +2,29 @@
 // Licensed under the MIT License.
 using System;
 using System.IO;
-using System.Linq;
 using Bicep.Cli.CommandLine.Arguments;
 
 namespace Bicep.Cli.CommandLine
 {
     public static class ArgumentParser
     {
-        public static ArgumentsBase? TryParse(string[] args)
+        public static (string?, ArgumentsBase?) TryParse(string[] args)
```

Comment:
>(string?, ArgumentsBase?) [](http://example.com/codeflow?start=22&length=25)

It looks like the tuple members are either both null or both non-null. You could simplify this to: `(string, ArgumentsBase)?`

A better option would be to just add the name of the command to `ArgumentBase` properties. This way the return type would be even simpler.

Reply:
Thank you for the suggestion. The second option appears to simplify all more... I will take a look at it and make and update! 👍

Reply:
Sounds good. Everything else looks good btw!

Reply:
@majastrz I updated this PR with the option suggested. I think now is more readable... Having the `CommandName` in the `ArgumentsBase` properties allows all types of commands to make use of this property if they need it (as in the case of `BulidOrDecompileArguments`). Nevertheless, if you have another suggestion please tell me and I will update it 😄

Reply:
Looks great!